### ISSUE TITLE
Generated service implementations should use Promise instead of deprecated Future

### ISSUE THREAD

**Issue by MitchAman** [vert-x3/vertx-grpc#49]
When playing around with vertx-grpc I noticed a familiar warning regarding the deprecation of `io.vertx.core.Future` from the [3.8.0 release](https://github.com/vert-x3/wiki/wiki/3.8.0-Deprecations-and-breaking-changes).

Given the proto:
```proto
...

option java_package = "examples";

service Greeter {
    rpc SayHello (HelloRequest) returns (HelloResponse) {
    };
}

...
```
A method is generated in `GreeterGrpc.GreeterVertxImplBase` with the signature:
```java
public void sayHello(examples.HelloRequest request, io.vertx.core.Future<examples.HelloResponse> response)
```

The generated code should switch over to using the `io.vertx.core.Promise` interface.  Perhaps it's possible to generate a second method for each rpc in the proto which uses a `io.vertx.core.Promise` instead of `io.vertx.core.Future` to allow users to switch off of the deprecated code, without breaking other users that don't want to switch yet?  Not sure how feasible that would be.

**Comment by MitchAman** [comment]
Not sure if this is still relevant after just seeing [Remove usage of custom Vert.x API for gRPC #37](https://github.com/vert-x3/vertx-grpc/issues/37)

**Comment by vietj** [comment]
for 3.x we should indeed use promises whenever it is needed.

**Comment by vietj** [comment]
@pmlopes do we need changes in the current compiler we support for 3.x ?

**Comment by vietj** [comment]
@MitchAman we do have actually a contribution to support the model in 4 written in Java https://github.com/vert-x3/vertx-grpc/pull/45

**Comment by pmlopes** [comment]
@vietj yes, this will require a refactor to the c++ compiler.

**Comment by vietj** [comment]
@pmlopes perhaps we can just update the generator to generate SuppressWarnings annotations ?

**Comment by pmlopes** [comment]
Merged to 3.8
